File: Leetcode/compareVersion.java
public int compareVersion(String version1, String version2) throws NumberFormatException {
        
        String[] v1 = version1.split("\\.");
        String[] v2 = version2.split("\\.");
        int i = 0;
        while (i < v1.length || i < v2.length) {
            if (i < v1.length && i < v2.length) {
                int v1Num = Integer.parseInt(v1[i]);
                int v2Num = Integer.parseInt(v2[i]);
                if (v1Num < v2Num) {
                    return -1;
                } else if (v1Num > v2Num) {
                    return 1;
                } else {
                    i++;
                }
            } else if (i < v1.length) {
                if (Integer.parseInt(v1[i]) == 0) {
                    i++;
                } else {
                    return 1;
                }
            } else if (i < v2.length) {
                if (Integer.parseInt(v2[i]) == 0) {
                    i++;
                } else {
                    return -1;
                }
            }
        }
        return 0;
    }

File: Leetcode/compareVersion.swift
func compareVersion(version1: String, _ version2: String) -> Int {
    var v1 = version1.componentsSeparatedByString(".")
    var v2 = version2.componentsSeparatedByString(".")
    
    let len1 = v1.count
    let len2 = v2.count
    var i = 0
    while i < len1 && i < len2 {
        let v1Num = Int(v1[i])
        let v2Num = Int(v2[i])
        if v1Num < v2Num {
            return -1
        } else if v1Num > v2Num {
            return 1
        } else {
            i++
        }
    }
    
    while i < len1 {
        let v1Num = Int(v1[i])
        if v1Num == 0 {
            i++
        } else {
            return 1
        }
    }
    
    while i < len2 {
        let v2Num = Int(v2[i])
        if v2Num == 0 {
            i++
        } else {
            return -1
        }
    }
    return 0
}
